llmfoundry/models/layers/ffn.py

```diff
@@ -23,34 +29,58 @@ def __init__(
         d_model: int,
         expansion_ratio: int,
         fc_type: str = 'torch',
+        ffn_hidden_size: Optional[int] = None,
         device: Optional[str] = None,
         bias: bool = True,
     ):
         super().__init__()
-        fc_kwargs: dict[str, Any] = {
+        if ffn_hidden_size is not None:
+            log.info(ignore_expr_log_msg % (expansion_ratio, ffn_hidden_size))
+        else:
+            ffn_hidden_size = d_model * expansion_ratio
+            if ffn_hidden_size != int(ffn_hidden_size):
```

Comment:
does this imply that the type of `expansion_ratio` is wrong and it should really be a `float`? (I think yes)

Reply:
`expansion_ratio` can be type float or int 
eg if someone puts `expansion_ratio: 3.5`, then `ffn_hidden_size == int(ffn_hidden_size)` so we're good 
but if they put something ridiculous like 3.123, then it'll probably error.

Reply:
ya, i'm saying the type signature on this constructor is wrong cause it says `expansion_ratio: int`